### DISCUSSION TITLE
Bad Response from reverse geocode

### DISCUSSION THREAD

**Discussion by akki401** [osm-search/Nominatim#3531] [Q&A]
Hello team,
I have my own Nominatim server which is running on EC2.
I am getting below error(Screenshot attached). 
But When I print the lat, lon values it printed as floating values but not exponential.
I felt the reverse API internally converting the float number to exponential. Is there any configuration in server to rectify this issue?

![image](https://github.com/user-attachments/assets/4a1f0521-e302-4b77-8466-34fc39b8da23)

-- Edit by moderator --
Text in the image:
```
2024-09-03 16:22:44,152 [ERROR] Bad response:
{
  "error": {
    "code": 400,
    "message": "Floating-point number expected for parameter 'lon'"
  }
}

from [retracted]/reverse?lat=49.2816126&lon=-4.42e-05&format=jsonv2&accept-language=en' with status code: '400'
```

**Comment by mtmail** [comment]
Isn't it your software that creates the URL including '4.42e-05' ?

**Reply by akki401** [reply]
I think NO.
Before passing these coordinates to GET method of reverse as a payload I printed and it printing as floating only

**Reply by mtmail** [reply]
Can you search for `49.2816126` in your webserver logs? That can be Apache2 or nginx webserver, depending on your setup. The files are often `/var/log/apache2/access.log` or `/var/log/nginx/access.log`. That's what Nominatim received from your software. Do you see floating point longitude there?

**Reply by akki401** [reply]
I found the logs and it received in exponential format and printed.
**Interesting part here is it give the error only for "lon"**
Screenshot attached

![image](https://github.com/user-attachments/assets/13a66813-9a07-4471-8f06-bc0069143603)

**Reply by mtmail** [reply]
I see you're using the Python `requests` library to generate the URL. Python likes to print very small numbers in scientific notation. Converting them to strings first usually works.

```python
import requests

params = {
    "lat": 49.2816126,
    "lon": 0.0000442
}

req = requests.Request('GET', 'https://example.com', params=params)
print(req.prepare().url)

params = {
    "lat": "{:.10f}".format(49.2816126).rstrip('0'),
    "lon": "{:.10f}".format(0.0000442).rstrip('0')
}

req = requests.Request('GET', 'https://example.com', params=params)
print(req.prepare().url)
```
prints
```
https://example.com/?lat=49.2816126&lon=4.42e-05
https://example.com/?lat=49.2816126&lon=0.0000442
```

**Reply by akki401** [reply]
Big Thankyou @mtmail. Yes I understood the problem.
I will do as you said in previous comment
